File: minimum-depth-of-binary-tree/wrong.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

# leetcode.com boilerplate
class Solution:
    def minDepth(self, root: TreeNode) -> int:
        return min_depth(root)


def min_depth(tree):
    lefty, righty = 0, 0
    if tree.left is not None:
        lefty = min_depth(tree.left) + 1
    if tree.right is not None:
        righty = min_depth(tree.right) + 1

    return min(lefty, righty)

File: validate-binary-search-tree/wrong.cpp
#include <vector>

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

bool is_binary_search_tree(const TreeNode *root) {
    if (!root) {
        return true;
    }
    std::vector<const TreeNode*> stack{root};
    do {
        const TreeNode *const node = stack.back();
        stack.pop_back();
        if (node->left) {
            if (!(node->left->val < node->val)) {
                return false;
            }
            stack.push_back(node->left);
        }
        if (node->right) {
            if (!(node->val < node->right->val)) {
                return false;
            }
            stack.push_back(node->right);
        }
    } while (!stack.empty());

    return true;
}

class Solution {
public:
    bool isValidBST(TreeNode* root) {
        return is_binary_search_tree(root);
    }
};
